src/actions/profile/ProfileAction.js

```diff
@@ -0,0 +1,62 @@
+import axios from 'axios';
+import { toastSuccess, toastFailure } from '../../utils/toast';
+import 'react-toastify/dist/ReactToastify.css';
+import history from '../../utils/history';
+
+import { profileActionTypes } from '../actionTypes';
+import URLS from '../../urlConnections';
+
+export const fetchProfile = () => (dispatch) => {
+  dispatch({
+    type: profileActionTypes.VIEW_PROFILE_REQUEST,
+  });
+  return axios.get(`${URLS.BACKEND_URL}/api/profiles/${sessionStorage.getItem('username')}`, {
+    headers: {
+      'content-type': 'application/json',
+      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
+    },
+  })
+    .then((response) => {
+      dispatch({
+        type: profileActionTypes.VIEW_PROFILE_SUCCESS,
+        payload: response.data.profile,
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: profileActionTypes.VIEW_PROFILE_FAILED,
+        payload: error.response.data.profile.detail,
+      });
+      // eslint-disable-next-line no-unused-expressions
+      sessionStorage.getItem('isLoggedIn') === 'false' ? history.push('/login') : null;
```

Comment:
This line can better be implemented by avoiding the null. Nothing seems to be affected when it is called but the state of the application can be especially in production environments. An if statement would fit in well. For more information about this. You could go through this resource shared in this same comment. https://eslint.org/docs/rules/no-unused-expressions